### PR TITLE
Masq 196 nonce

### PR DESCRIPTION
**Pull request by leventdem**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -468,14 +482,18 @@ class Masq {
     if (!this.dataEncryptionKey) throw Error('Data encryption key is not set')
   }
 
+  _checkNonce () {
+    if (!this.nonce) throw Error('Nonce is not set')
+  }
```
Comment (sujeebant) [inline src/index.js]: function _checkNonce and _checkDEK never called ?

File: src/index.js [hunk 2]
```diff
@@ -247,6 +249,9 @@ class Masq {
         if (!json.userAppDEK) {
```
Comment (G-Ray) [inline src/index.js]: weh have many check like this, we can reduce LOCs by iterating on an array of properties to validate

File: src/index.js [hunk 3]
```diff
@@ -468,14 +482,18 @@ class Masq {
...
   /**
    * Set a watcher
    * @param {string} key - Key
    * @returns {Object}
    */
-  watch (key, cb) {
+  async watch (key, cb) {
     const db = this._getDB()
-    return db.watch(key, cb)
+    await common.utils.watch(db, this.nonce, key, cb)
```
Comment (G-Ray) [inline src/index.js]: I don't think watch is an async function

File: src/index.js (hunk 1 above)
Comment (leventdem) [reply src/index.js]: yes removed.

File: src/index.js (hunk 3 above)
Comment (leventdem) [reply src/index.js]: yes but we are calling hashKey inside watch method, and this call is async.

Comment (G-Ray) [reply src/index.js]: ok but this will be proceesed inside the callback IIUC ?

Comment (sujeebant) [reply src/index.js]: I don't understand what callback IIUC is ?
But the hashKey function is called before the actual 'not async' method watch on hyperdb

Comment (G-Ray) [reply src/index.js]: lol IIUC stand for "If I Understand Correctly" :rofl:

Comment (G-Ray) [reply src/index.js]: that's true, we need to wait for `hashKey`, so it seems we need to await this watch method

**Pull request merged**